src/commands/lint/configureLintingToolsCommand.ts

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { commands, l10n, window, ExtensionContext } from 'vscode';
+import * as fs from 'fs';
+import * as path from 'path';
+import { WorkspaceUtils } from '../../utils/workspaceUtils';
+import { TAB_SPACES } from '../../utils/constants';
+
+const configureLintingToolsCommand =
+    'salesforcedx-vscode-offline-app.configureLintingTools';
+const eslintDependencies = [
+    ['@salesforce/eslint-plugin-lwc-graph-analyzer', '^0.9.0'],
+    ['eslint', '^8.47.0']
+];
+const lwcGraphAnalyzerRecommended: string =
+    'plugin:@salesforce/lwc-graph-analyzer/recommended';
+const eslintRecommended = 'eslint:recommended';
+
+interface PackageJson {
+    devDependencies?: Record<string, string>;
+}
+
+enum MessageType {
+    Error,
+    InformationYesNo,
+    InformationOk
+}
+
+export class ConfigureLintingToolsCommand {
+    static async configure(): Promise<boolean> {
+        try {
+            if (!WorkspaceUtils.lwcFolderExists()) {
+                await this.showMessage('LWC folder does not exist.');
+                return Promise.resolve(false);
+            }
+
+            if (!WorkspaceUtils.packageJsonExists()) {
+                await this.showMessage(
+                    'The project does not contain package.json.'
+                );
+                return Promise.resolve(false);
+            }
+
+            // Ask user to add eslint plugin
+            const result = await this.showMessage(
+                'Do you want to add eslint plugin for LWC data graph anaylsis to your package.json?',
+                MessageType.InformationYesNo
+            );
+
+            if (!result || result.title === l10n.t('No')) {
+                return Promise.resolve(false);
+            } else {
+                let modified = false;
+                try {
+                    modified = this.updateDevDependencies();
+                } catch (error) {
+                    await this.showMessage(
+                        `Error updating package.json: ${error}`
+                    );
+                    return Promise.resolve(false);
+                }
+
+                try {
+                    this.updateEslintrc();
```

Comment:
We should be tracking whether this file was updated or not too, and reporting to the user accordingly—even if the report is built into a single message. We could be updating this but not the `devDependencies`, and vice versa. We should be clear to the user, exactly what's being updated and what is not.

Reply:
Showing 3 messages now at a maximum, one for updating `devdependencies`, one for updating `.eslintrc.json`, and one for running npm. I thought maybe I should mash the three. But localizing string was going to be tricky so instead went with three.

![image](https://github.com/salesforce/salesforcedx-vscode-mobile/assets/42476999/9be4e21d-ba87-450d-8374-06810d478f19)